core/src/commonMain/kotlin/supplier/RestEntitySupplier.kt

```diff
@@ -647,6 +648,38 @@ public class RestEntitySupplier(public val kord: Kord) : EntitySupplier {
         GuildApplicationCommand(data, interaction)
     }
 
+    override suspend fun getEntitlementOrNull(applicationId: Snowflake, entitlementId: Snowflake): Entitlement? = catchNotFound {
+        val response = entitlement.getEntitlement(applicationId, entitlementId)
+        val data = EntitlementData.from(response)
+        Entitlement(data, kord)
+    }
+
+    override suspend fun getEntitlements(
+        applicationId: Snowflake,
+        skuId: Snowflake?,
+        limit: Int?,
+        userId: Snowflake?,
+        guildId: Snowflake?
+    ): Flow<Entitlement> = limitedPagination(limit, maxBatchSize = 100) { batchSize ->
+        paginateForwards(batchSize, idSelector = { it.id }) { position ->
+            entitlement.listEntitlements(
+                applicationId = applicationId,
+                request = EntitlementsListRequest(
+                    position = position,
+                    limit = batchSize,
+                    skuIds = listOfNotNull(skuId),
+                    userId = userId,
+                    guildId = guildId
+                )
+            )
+        }.map {
+            val data = EntitlementData.from(it)
+            Entitlement(data, kord)
+        }
+    }
+
+    public suspend fun getSKUs(applicationId: Snowflake): List<Sku> =
+        sku.listSkus(applicationId).map { Sku(it, kord) }
```

Comment:
i think this should be directly inlined into `Kord` instead, see `getApplicationRoleConnectionMetadataRecords`